tools/clang/unittests/HLSL/ExecutionTest.cpp

```diff
@@ -9776,51 +9783,60 @@ TEST_F(ExecutionTest, DynamicResourcesDynamicIndexingTest) {
+  // unsigned num_models_to_test = ExecutionTest::IsFallbackPathEnabled() ? y : x;
+  unsigned num_models_to_test = ExecutionTest::IsFallbackPathEnabled() ? 2 : 1;
+  for (unsigned i = 0; i < num_models_to_test; i++) {
+    D3D_SHADER_MODEL sm = TestShaderModels[i];
+    LogCommentFmt(L"\r\nVerifying Dynamic Resources Uniform Indexing in shader "
+                  L"model 6.%1u",
+                  ((UINT)sm & 0x0f));
 
-      CComPtr<ID3D12Device> pDevice;
-      if (!CreateDevice(&pDevice, sm, false /* skipUnsupported */)) {
-        continue;
-      }
-      D3D12_FEATURE_DATA_D3D12_OPTIONS devOptions;
-      VERIFY_SUCCEEDED(
-          pDevice->CheckFeatureSupport((D3D12_FEATURE)D3D12_FEATURE_D3D12_OPTIONS,
-                                       &devOptions, sizeof(devOptions)));
-      if (devOptions.ResourceBindingTier < D3D12_RESOURCE_BINDING_TIER_3) {
-        WEX::Logging::Log::Comment(
-            L"Device does not support Resource Binding Tier 3");
-        WEX::Logging::Log::Result(WEX::Logging::TestResults::Skipped);
-        return;
-      }
+    CComPtr<ID3D12Device> pDevice;
+    if (!CreateDevice(&pDevice, sm, false /* skipUnsupported */)) {
+      continue;
+    }
+    D3D12_FEATURE_DATA_D3D12_OPTIONS devOptions;
+    VERIFY_SUCCEEDED(
+        pDevice->CheckFeatureSupport((D3D12_FEATURE)D3D12_FEATURE_D3D12_OPTIONS,
+                                      &devOptions, sizeof(devOptions)));
+    if (devOptions.ResourceBindingTier < D3D12_RESOURCE_BINDING_TIER_3) {
+      WEX::Logging::Log::Comment(
+          L"Device does not support Resource Binding Tier 3");
+      WEX::Logging::Log::Result(WEX::Logging::TestResults::Skipped);
+      return;
+    }
 
-      // Add compile options
+    for (unsigned int non_uniform_bit = 0; non_uniform_bit < 2; non_uniform_bit++) {
+      float *expectedResults = non_uniform_bit ? expectedResultsNonUniform : expectedResultsUniform;
+
+      LogCommentFmt(L"\r\nnon uniform bit is %1u",
+                  non_uniform_bit);
 
+      // Add compile options
       std::string compilerOptions = "";
       if (sm==D3D_SHADER_MODEL_6_0)
-        compilerOptions += "-D FALLBACK=1";
-      if (non_uniform_bit && sm==D3D_SHADER_MODEL_6_0)
-        compilerOptions += ", -D NON_UNIFORM=1";
-      else if (non_uniform_bit)
-        compilerOptions += "-D NON_UNIFORM=1";
+        compilerOptions += " -D FALLBACK=1";
+      if (non_uniform_bit)
+        compilerOptions += " -D NON_UNIFORM=1";
 
       // by default a root value is added.
       // remove the root value if this is the non-fallback path
-      if (sm==D3D_SHADER_MODEL_6_6 && non_uniform_bit)
+      if (sm==D3D_SHADER_MODEL_6_6)
       {
         pShaderOp->RootValues.clear();
       }
+      else
+      {
+         pShaderOp->RootValues = oldRootValues;
+      }
 
       // Update shader target in xml.
       for (st::ShaderOpShader &S : pShaderOp->Shaders){
+        S.Arguments = NULL;
         if (compilerOptions != ""){
```

Comment:
`!compilerOptions.empty()` should also work from default initialized `std::string`.